src/libraries/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableArray_1.Builder.cs

```diff
@@ -438,6 +537,91 @@ public void RemoveAt(int index)
                 this.Count--;
             }
 
+            /// <summary>
+            /// Removes the specified values from this list.
+            /// </summary>
+            /// <param name="index">The 0-based index into the array for the element to omit from the returned array.</param>
+            /// <param name="length">The number of elements to remove.</param>
+            public void RemoveRange(int index, int length)
+            {
+                Requires.Range(index >= 0 && index + length <= _count, nameof(index));
+
+                if (length == 0)
+                {
+                    return;
+                }
+
+                if (index + length < this._count)
+                {
+                    Array.Copy(_elements, index + length, _elements, index, this.Count - index - length);
+                }
```

Comment:
For reference types, we'd still need to clear any unused elements from the buffer for reclamation by the GC. See the `List<T>` implementation for an example:

https://github.com/dotnet/runtime/blob/main/src/libraries/System.Private.CoreLib/src/System/Collections/Generic/List.cs#L905-L908